public/app/features/dashboard/state/initDashboard.ts

```diff
@@ -0,0 +1,227 @@
+        return getNewDashboardModelData(args.urlFolderId);
+      }
+    }
+  } catch (err) {
+    dispatch(setDashboardLoadingState(DashboardLoadingState.Error));
+    dispatch(notifyApp(createErrorNotification('Dashboard fetch failed', err)));
+    console.log(err);
+    return null;
+  }
+}
+
+/**
+ * This action (or saga) does everything needed to bootstrap a dashboard & dashboard model.
+ * First it handles the process of fetching the dashboard, correcting the url if required (causing redirects/url updates)
+ *
+ * This is used both for single dashboard & solo panel routes, home & new dashboard routes.
+ *
+ * Then it handles the initializing of the old angular services that the dashboard components & panels still depend on
+ *
+ */
+export function initDashboard(args: InitDashboardArgs): ThunkResult<void> {
+  return async (dispatch, getState) => {
+    // set fetching state
+    dispatch(setDashboardLoadingState(DashboardLoadingState.Fetching));
+
+    // Detect slow loading / initializing and set state flag
+    // This is in order to not show loading indication for fast loading dashboards as it creates blinking/flashing
+    setTimeout(() => {
+      if (getState().dashboard.model === null) {
+        dispatch(setDashboardLoadingSlow());
+      }
+    }, 500);
+
+    // fetch dashboard data
+    const dashDTO = await fetchDashboard(args, dispatch, getState);
+
+    // returns null if there was a redirect or error
+    if (!dashDTO) {
+      return;
+    }
+
+    // set initializing state
+    dispatch(setDashboardLoadingState(DashboardLoadingState.Initializing));
+
+    // create model
+    let dashboard: DashboardModel;
+    try {
+      dashboard = new DashboardModel(dashDTO.dashboard, dashDTO.meta);
+    } catch (err) {
+      dispatch(setDashboardLoadingState(DashboardLoadingState.Error));
+      dispatch(notifyApp(createErrorNotification('Dashboard model initializing failure', err)));
+      console.log(err);
+      return;
+    }
+
+    // add missing orgId query param
+    if (!getState().location.query.orgId) {
+      dispatch(updateLocation({ query: { orgId: config.bootData.user.orgId }, partial: true, replace: true }));
+    }
+
+    // init services
+    const timeSrv: TimeSrv = args.$injector.get('timeSrv');
+    const annotationsSrv: AnnotationsSrv = args.$injector.get('annotationsSrv');
+    const variableSrv: VariableSrv = args.$injector.get('variableSrv');
+    const keybindingSrv: KeybindingSrv = args.$injector.get('keybindingSrv');
+    const unsavedChangesSrv = args.$injector.get('unsavedChangesSrv');
+    const dashboardSrv: DashboardSrv = args.$injector.get('dashboardSrv');
+
+    timeSrv.init(dashboard);
+    annotationsSrv.init(dashboard);
+
+    // template values service needs to initialize completely before
+    // the rest of the dashboard can load
+    try {
+      await variableSrv.init(dashboard);
+    } catch (err) {
+      dispatch(notifyApp(createErrorNotification('Templating init failed', err)));
```

Comment:
Would it be an idea to pass `(err.statusText || err.message || undefined)` instead of err to avoid `[object Object]`?

![screenshot 2019-02-06 12 52 37](https://user-images.githubusercontent.com/224493/52339865-92a97a80-2a0e-11e9-9fd7-521f8e7c6f86.png)